app/src/androidTest/java/net/skyscanner/backpack/bottomsheet/BpkBottomSheetTest.kt

```diff
@@ -18,44 +18,135 @@
 
 package net.skyscanner.backpack.bottomsheet
 
+import android.view.View
 import android.widget.FrameLayout
+import androidx.appcompat.app.AppCompatActivity
 import androidx.appcompat.content.res.AppCompatResources
 import androidx.coordinatorlayout.widget.CoordinatorLayout
 import androidx.coordinatorlayout.widget.CoordinatorLayout.LayoutParams
+import androidx.test.espresso.Espresso
+import androidx.test.espresso.IdlingRegistry
+import androidx.test.espresso.IdlingResource
+import androidx.test.espresso.IdlingResource.ResourceCallback
+import androidx.test.espresso.matcher.ViewMatchers.assertThat
+import androidx.test.espresso.matcher.ViewMatchers.withId
+import androidx.test.ext.junit.rules.activityScenarioRule
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import androidx.test.platform.app.InstrumentationRegistry
+import com.google.android.material.bottomsheet.BottomSheetBehavior
+import com.google.android.material.bottomsheet.BottomSheetBehavior.BottomSheetCallback
+import com.google.android.material.bottomsheet.BottomSheetBehavior.STATE_EXPANDED
 import net.skyscanner.backpack.BpkSnapshotTest
+import net.skyscanner.backpack.BpkTestVariant
 import net.skyscanner.backpack.demo.R
+import org.hamcrest.CoreMatchers.`is`
+import org.hamcrest.Matchers.greaterThanOrEqualTo
+import org.hamcrest.Matchers.lessThanOrEqualTo
 import org.junit.Before
+import org.junit.Rule
 import org.junit.Test
 import org.junit.runner.RunWith
 
 @RunWith(AndroidJUnit4::class)
 class BpkBottomSheetTest : BpkSnapshotTest() {
 
-  private val root = CoordinatorLayout(testContext)
+  @get:Rule
+  val rule = activityScenarioRule<AppCompatActivity>()
 
-  init {
+  private lateinit var bottomSheetBehaviour: BpkBottomSheetBehaviour<BpkBottomSheet>
+
+  @Before
+  fun setup() {
+    setDimensions(200, 200)
+  }
+
+  @Test
+  fun default() {
+    snap(setupBottomSheet())
+  }
+
+  @Test
+  fun expanded() {
```

Comment:
testing this is mighty annoying as we need to wait for it to have changed the state